Lib/xml/etree/ElementTree.py

```diff
@@ -1248,10 +1248,16 @@ def iterator(source):
             if close_source:
                 source.close()
 
+    gen = iterator(source)
     class IterParseIterator(collections.abc.Iterator):
-        __next__ = iterator(source).__next__
+        __next__ = gen.__next__
+        def close(self):
+            if close_source:
+                source.close()
+            gen.close()
 
         def __del__(self):
+            # TODO: Emit a ResourceWarning if it was not explicitly closed.
```

Comment:
Any plans for when this TODO will be done? They can end up languishing for decades :)

Reply:
When all maintained Python versions have the `close()` method, so you can call it without checking the version.

Reply:
Maybe update the TODO to mention this, so when someone sees it they know whether they can do the TODO or wait a bit longer?

Anyway, I'll approve this.